common/basecomponent/libp2p.go

```diff
@@ -0,0 +1,24 @@
+package basecomponent
```

Comment:
Why is this a separate file? It's used in only 1 place, and we already have a bunch of other P2P stuff in this package. If it needs to be separate, please move it to `/cmd/dendrite-p2p-demo/libp2p.go` instead.

Reply:
Done in 6291732